xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/less/general.less

```
@@ -168,13 +177,21 @@ body.content {
       & .buttonwrapper,
       & #user-menu-col {
         & a {
-          text-decoration: none;
-          // Make sure we don't take over the default behaviour on hover with too much specificity.
-          &:hover, &:focus {
-            text-decoration: underline
-          }
+          .not-inline-link();
         }
       }
+
+      /* Remove the underlining from links with the btn class.
+       Those are already styled differently from their neighboring text. */
+      & a.btn {
+        .not-inline-link();
+      }
+
+      // Remove the underlining from links with an icon. In the vast majority of cases, these are already spottable.
+      & a:has( img[alt=""]), // With the silk iconTheme
+      & a:has( span.fa) { // With the Font Awesome iconTheme
```

```suggestion
      & a:has(span.fa) { // With the Font Awesome iconTheme
```

Also, at least in Java, we put comments on extra lines, I don't know if such a rule also exists for CSS?

---

Change made in https://github.com/xwiki/xwiki-platform/pull/3077/commits/0764066aace1eadc145603f45d2aa7fe312148c2 (with a change on the line above too).

> Also, at least in Java, we put comments on extra lines, I don't know if such a rule also exists for CSS?

We have [some commenting code style for CSS](https://dev.xwiki.org/xwiki/bin/view/Community/CodeStyle/XhtmlCssCodeStyle/#HCSS), but AFAIK they don't force a position for the comment. They are not even respected in this LESS file. I kept the commenting style already here.

About it being on the exact line, I think it's useful here because each comment is meant to describe exactly the line it's on. They wouldn't make sense at the level of the ruleset, that's why I separated them from the `//Remove the underlining from [...]` comment.